Jenkinsfile

```diff
@@ -12,69 +12,103 @@ spec:
+    volumeMounts:
+      - name: settings-xml
+        mountPath: /home/jenkins/.m2/settings.xml
+        subPath: settings.xml
+        readOnly: true
+      - name: m2-repo
+        mountPath: /home/jenkins/.m2/repository
     resources:
       limits:
-        memory: "4Gi"
+        memory: "8Gi"
         cpu: "2"
       requests:
-        memory: "4Gi"
+        memory: "8Gi"
         cpu: "2"
+  volumes:
+    - name: settings-xml
+      secret:
+        secretName: m2-secret-dir
+        items:
+        - key: settings.xml
+          path: settings.xml
+    - name: m2-repo
+      emptyDir: {}
 """
     }
   }
-    stages {
-        stage('repairnator-core'){
-            environment {
-                TEST_PATH="src/repairnator-core/"
-            }
-            steps {
-                container('maven') {
-                  sh 'bash ./.ci/ci-run.sh'
-                }
-            }
+  stages {
+    stage('repairnator-core'){
+      environment {
+          TEST_PATH="src/repairnator-core/"
+      }
+      steps {
+        container('maven') {
+          sh 'bash ./.ci/ci-run.sh'
         }
-        stage('repairnator-pipeline'){
-            environment {
-                TEST_PATH="src/repairnator-pipeline/"
-                TEST_LIST="**"
-            }
-            steps {
-                container('maven') {
-                  sh 'bash ./.ci/ci-run-with-core.sh'
-                }
-            }
+      }
+      options {
+          timeout(time: 15, unit: "MINUTES")
+      }
+    }
+    stage('repairnator-realtime'){
+      environment {
+        TEST_PATH="src/repairnator-realtime/"
+        TEST_LIST="**"
+      }
+      steps {
+        container('maven') {
+          sh 'bash ./.ci/ci-run-with-core.sh'
         }
-        stage('repairnator-realtime'){
-            environment {
-                TEST_PATH="src/repairnator-pipeline/"
-                TEST_LIST="**"
-            }
-            steps {
-                container('maven') {
-                  sh 'bash ./.ci/ci-run-with-core.sh'
-                }
-            }
+      }
+      options {
+        timeout(time: 15, unit: "MINUTES")
+      }
+    }
+    stage('repairnator-jenkins-plugin'){
+      environment {
+          TEST_PATH="src/repairnator-jenkins-plugin/"
+      }
+      steps {
+        container('maven') {
+          sh 'bash ./.ci/ci-run.sh'
         }
-        stage('repairnator-jenkins-plugin'){
-            environment {
-                TEST_PATH="src/repairnator-jenkins-plugin/"
-            }
-            steps {
-                container('maven') {
-                  sh 'bash ./.ci/ci-run.sh'
-                }
-            }
+      }
+      options {
+        timeout(time: 15, unit: "MINUTES")
+      }
+    }
+    stage('repairnator-maven-repair'){
+      environment {
+        TEST_PATH="src/maven-repair/"
+        TEST_LIST="**"
+      }
+      steps {
+        container('maven') {
+          sh 'bash ./.ci/ci-maven-repair.sh'
         }
```

Comment:
Is there really a need to use a separate script for `maven-repair`? It would be nice to merge this with `run-with-core`. The only difference is the `NPEFix` version option.

If we change this we can eliminate `xmlstarlet` from the Docker image too.

Reply:
I'm not really sure of what does that version property does, have you tried to run the tests without it?

Reply:
Yes, and they work. I also don't know why that is needed. I haven't found any usage of a property `NPEFIX_VERSION`.

Reply:
@cokefh @monperrus WDYT?